app/controllers/users_controller.rb

```diff
@@ -13,16 +21,64 @@ def new
   def create
     @user = User.new user_params
     if @user.save
-      flash[:success] = t :welcome
+      log_in @user
+      flash[:success] = t ".welcome"
       redirect_to @user
     else
       render :new
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      redirect_to root_path
+    end
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      redirect_to root_path
+    else
+      if @user.update_attributes user_params
+        flash[:success] = t ".profile_updated"
+        redirect_to @user
+      else
+        render :edit
+      end
+    end
+  end
+
+  def destroy
+    user = User.find_by(id: params[:id])
+    unless user.nil?
+      user.destroy
+      flash[:success] = t ".user_deleted"
+    end
+    redirect_to users_url
```

Comment:
So the request will be redirected to users_url for both of the case(delete/not deleted), it that right?

Reply:
@mdmahmudurrahman So let it doesn't redirect ?